distribution/lib/Standard/Base/0.0.0-dev/src/Data/Array_Proxy.enso

```diff
@@ -33,4 +33,4 @@ type Array_Proxy
        methods.
     from_proxy_object : Any -> Array
     from_proxy_object proxy =
-        Array_Proxy.new proxy.length (i -> proxy.at i)
+        Array_Proxy.new proxy.length proxy.at
```

Comment:
Don't we want to keep this until curried is +- as fast as lambda? Or is the plan to fix this as part of this PR / a follow up one soon?

Reply:
Yes, the plan is to fix it by changes in this PR. In fact it has already been fixed. Running yields:
```
sbt:runtime> withDebug benchOnly -- averageOverArrayProxy
averageOverArrayProxy - 0.010 ms/op
averageOverArrayProxyNew - 0.014 ms/op
```
now the curried variant is even faster.

Reply:
I fixed the PR title and descriptions. Need your approval @radeusgd and @jdunkerley thanks to this change in standard library.